libdrgn/kdump_reader.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 - Serapheim Dimitropoulos
+// SPDX-License-Identifier: GPL-3.0+
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "kdump_reader.h"
+#include "program.h"
+
+bool has_kdump_signature(int fd, struct drgn_error **err)
```

Comment:
The calling convention in drgn is that if an error may be returned, the error is always the return value, and anything else is returned through a return parameter.